### ISSUE TITLE
Request returns 405 error!

### ISSUE THREAD

**Issue by renevt** [ohld/igbot#756]
## Please follow the guide below

- Issues submitted without this template format will be **ignored**.
- You will be asked some questions and requested to provide some information, please read them **carefully** and answer completely.
- Put an `x` into all the boxes [ ] relevant to your issue (like so [x]).
- Use the *Preview* tab to see how your issue will actually look like.

---

### Before submitting an issue, make sure you have:
- [x ] Updated to the latest version
- [x ] Read the [README](https://github.com/instagrambot/instabot/blob/master/README.md) and [docs](https://github.com/instagrambot/instabot/tree/master/docs)
- [x ] [Searched](https://github.com/instagrambot/instabot/search?type=Issues) the bugtracker for similar issues including **closed** ones

### Purpose of your issue?
- [x ] Bug report (encountered problems/errors)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### The following sections requests more details for particular types of issues, you can remove any section (the contents between the triple ---) not applicable to your issue.

---

### For a *bug report*, you **must** include the Python version used, *code* that will reproduce the error, and the *error log/traceback*.

Paste the output of ``python -V``
:

Code:  Python 2.7.12

`` python like_hashtags.py *******
2019-01-17 10:34:31,202 - INFO - Instabot Started
Which account do you want to use? (Type number)
1: ********
2: ********
0: add another account.
-1: delete all accounts.
2
2019-01-17 10:34:36,600 - INFO - Logged-in successfully as '*****.nl' using the cookie!
2019-01-17 10:34:36,601 - INFO - Going to like media with hashtag #*********.
2019-01-17 10:34:38,642 - INFO - Going to like 77 medias.
  0%|                                                    | 0/77 [00:00<?, ?it/s]2019-01-17 10:34:38,792 - ERROR - Request returns 405 error!

Media ID error!
^C
Traceback (most recent call last):
  File "like_hashtags.py", line 27, in <module>
    bot.like_hashtag(hashtag)
  File "/usr/lib/python2.7/site-packages/instabot/bot/bot.py", line 431, in like_hashtag
    return like_hashtag(self, hashtag, amount)
  File "/usr/lib/python2.7/site-packages/instabot/bot/bot_like.py", line 92, in like_hashtag
    return self.like_medias(medias)
  File "/usr/lib/python2.7/site-packages/instabot/bot/bot.py", line 419, in like_medias
    return like_medias(self, media_ids, check_media)
  File "/usr/lib/python2.7/site-packages/instabot/bot/bot_like.py", line 53, in like_medias
    self.error_delay()
  File "/usr/lib/python2.7/site-packages/instabot/bot/bot.py", line 272, in error_delay
    time.sleep(10)
KeyboardInterrupt
2019-01-17 10:34:43,064 - INFO - Bot stopped. Worked: 0:04:23.970103
2019-01-17 10:34:43,065 - INFO - Total requests: 9

---

### Describe your issue

Hi

Since a day or two i got this error, the bot wouldnt like anything anymore . Follow, unfollow ,comment and etc. works normaly.

**Comment by Rouxcoucou** [comment]
I have the same error since yesterday night...

**Comment by stas2z** [comment]
Same since yesterday evening, it stops at media_info
seems like media/{id}/info api request is forbidden (405) atm

**Comment by JULIANCRACK** [comment]
same here since morning (paris time)

**Comment by sh0ked** [comment]
same here since morning (+7 UTC)

**Comment by ohld** [comment]
@maxdevblock @sudoguy I think we have broken something. Please check this ASAP.

**Comment by maxdevblock** [comment]
> @maxdevblock @sudoguy I think we have broken something. Please check this ASAP.

Don't think it's an instabot issue. I got same error with older versions.
Instagram has currently got an outage issue in US and EU.

**Comment by ohld** [comment]
Have we somehow tackle this? Maybe add some description message in instabot or some error handling to be more descriptive and less frightened?

**Comment by maxdevblock** [comment]
> Have we somehow tackle this? Maybe add some description message in instabot or some error handling to be more descriptive and less frightened?

I'm updating users on telegram group

**Comment by maxdevblock** [comment]
Seems the problem is only with
`'media/{media_id}/info/'.format(media_id=media_id)`
requests.

**Comment by ohld** [comment]
So what is the best way to do having this problem?

**Comment by maxdevblock** [comment]
Ok. Got the problem. Don't know how to solve (if we can solve or have to wait).
The call `'media/{media_id}/info/'.format(media_id=media_id)`
is returning only
```
{'logged_in_user': { ALL_INFO_ABOUT_USER },
 'status': 'ok'}
```
but not `items` so the def `get_media_info` is failing.

**Comment by maxdevblock** [comment]
I'm curretly running a test every 30 min. If something changes I'll write here and in telegram

**Comment by barclaey** [comment]
Hi guys,

I think I found the problem.

In api.py change media_info(self, media_id) to:
```python
    def media_info(self, media_id):
        url = 'media/{media_id}/info/'.format(media_id=media_id)
        return self.send_request(url)
```

That seems to do the trick.

**Comment by lukasz00500** [comment]
I have The same issue, but old version of this bot work good

**Comment by maxdevblock** [comment]
@barclaey you're right! It works without sending json_data... :scream:

**Comment by maxdevblock** [comment]
@barclaey it's working. I can't get the reason why... But working!
Send the pull request

**Comment by silvio2810** [comment]
@maxdevblock I've tried your trick, and for me it doesn't work as well. :(

**Comment by maxdevblock** [comment]
@silvio2810 sent a pull req now with other fixes.

**Comment by silvio2810** [comment]
@maxdevblock 

def media_info(self, media_id):
        url = 'media/{media_id}/info/'.format(media_id=media_id)
        return self.send_request(url, '')

I've seen that pull, and I'm testing, but nothing!

**Comment by ohld** [comment]
What else can we try?

**Comment by maxdevblock** [comment]
@silvio2810 no... @barclaey was right
`return self.send_request(url)`
But I fixed tests too cause in tests the req was POST, now it's GET. With fixed tests, Travis doesn't fail and pull req can pass. 
#758

**Comment by silvio2810** [comment]
I've seen the tests and the change that you made from POST to GET. I've tested it as well but nothing changed. I'm sorry to go against you @maxdevblock and @barclaey. :'(

**Comment by maxdevblock** [comment]
@silvio2810 

It's working for me, both python2 and 3.

did u install instabot via pip or git clone?

If u installed via:
- git: edit local files
- pip: edit sys files
- both: edit sys files, not local...

To know were your sys instabot package is,
`python -c "import instabot; print(instabot.__file__)"`

**Comment by renevt** [comment]
@maxdevblock

This fix doesnt work for me either.
Mabye you can help me out.

 pip install instabot
Requirement already satisfied: instabot in /usr/lib/python2.7/site-packages (0.4.5)
Requirement already satisfied: tqdm in /usr/lib/python2.7/site-packages (from instabot) (4.28.1)
Requirement already satisfied: requests-toolbelt in /usr/lib/python2.7/site-packages (from instabot) (0.8.0)
Requirement already satisfied: requests in /usr/lib/python2.7/site-packages/requests-2.21.0-py2.7.egg (from instabot) (2.21.0)
Requirement already satisfied: schedule in /usr/lib/python2.7/site-packages (from instabot) (0.5.0)
Requirement already satisfied: future in /usr/lib/python2.7/site-packages (from instabot) (0.17.1)
Requirement already satisfied: six in /usr/lib/python2.7/site-packages (from instabot) (1.12.0)
Requirement already satisfied: huepy in /usr/lib/python2.7/site-packages (from instabot) (0.9.8.1)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in /usr/lib/python2.7/site-packages/chardet-3.0.4-py2.7.egg (from requests->instabot) (3.0.4)
Requirement already satisfied: idna<2.9,>=2.5 in /usr/lib/python2.7/site-packages/idna-2.8-py2.7.egg (from requests->instabot) (2.8)
Requirement already satisfied: urllib3<1.25,>=1.21.1 in /usr/lib/python2.7/site-packages/urllib3-1.24.1-py2.7.egg (from requests->instabot) (1.24.1)
Requirement already satisfied: certifi>=2017.4.17 in /usr/lib/python2.7/site-packages/certifi-2018.11.29-py2.7.egg (from requests->instabot) (2018.11.29)

 python -c "import instabot; print(instabot.__file__)"
/usr/lib/python2.7/site-packages/instabot/__init__.pyc


    def remove_self_tag(self, media_id):
        data = self.json_data()
        url = 'media/{media_id}/remove/'.format(media_id=media_id)
        return self.send_request(url, data)

    def media_info(self, media_id):
        url = 'media/{media_id}/info/'.format(media_id=media_id)
        return self.send_request(url)

    def archive_media(self, media, undo=False):
        action = 'only_me' if not undo else 'undo_only_me'
        data = self.json_data({'media_id': media['id']})
        url = 'media/{media_id}/{action}/?media_type={media_type}'.format(
            media_id=media['id'],
            action=action,
            media_type=media['media_type']

I'm not that familair with python and linux.

Thank you

**Comment by maxdevblock** [comment]
@renevt are you editing
`/usr/lib/python2.7/site-packages/instabot/api/api.py`
?

**Comment by renevt** [comment]
@maxdevblock hmmmz let me think
im not sure let me figure it out

**Comment by maxdevblock** [comment]
@renevt 
ok try to run a script .py
```
from instabot import Bot

bot = Bot()
bot.login()

media_id = '1957463722880884083'
pk = bot.get_media_info(media_id)[0]['pk']
bot.logger.warn("\033[1;32m{}\033[0m".format(media_id == str(pk)))
```

If the green warn is `True` the fix is working

**Comment by renevt** [comment]
@maxdevblock 

The cookie is not found, but don't worry `instabot` will create it for you using your login details.
2019-01-18 08:49:38,440 - INFO - Logged-in successfully as '*******'!
2019-01-18 08:49:38,442 - INFO - Saved cookie!
2019-01-18 08:49:38,672 - WARNING - True (green ;))
2019-01-18 08:49:38,898 - INFO - Bot stopped. Worked: 0:00:23.470825
2019-01-18 08:49:38,899 - INFO - Total requests: 4
administrator@DS216play://volume1/web$

**Comment by maxdevblock** [comment]
@renevt 
WORKING :-)

**Comment by renevt** [comment]
@maxdevblock 
2019-01-18 08:52:20,877 - INFO - Going to like media with hashtag #****.
2019-01-18 08:52:24,068 - INFO - Going to like 81 medias.
  0%|                                                    | 0/81 [00:00<?, ?it/s]2019-01-18 08:52:24,220 - ERROR - Request returns 405 error!

Media ID error!
  1%|▌                                           | 1/81 [00:10<13:32, 10.16s/it]2019-01-18 08:52:34,377 - ERROR - Request returns 405 error!

Media ID error!


Really strange im such a noob why does it not work

**Comment by maxdevblock** [comment]
@renevt send me the first lines of the script you're using, from first line to `bot = Bot()`

**Comment by renevt** [comment]
@maxdevblock 

im using the multi_script_CLI script for most of the times

import getpass
import os
import random
import sys
import time


from tqdm import tqdm

sys.path.append(os.path.join(sys.path[0], '../'))
from instabot import Bot



bot = Bot(filter_business_accounts=False)
bot.login()


Thank you

**Comment by maxdevblock** [comment]
@renevt ok... 
comment out
`sys.path.append(os.path.join(sys.path[0], '../'))`
to
`# sys.path.append(os.path.join(sys.path[0], '../'))`
this line is overriding import from system installed module and importing the local one.

**Comment by renevt** [comment]
@maxdevblock 

No sorry i doesnt work
2019-01-18 09:05:33,979 - INFO - Instabot Started
Which account do you want to use? (Type number)
1: ****
0: add another account.
-1: delete all accounts.
1
2019-01-18 09:05:38,916 - INFO - Logged-in successfully as '******' using the cookie!
2019-01-18 09:05:38,919 - INFO - Instabot Started
Which account do you want to use? (Type number)
1: ****
0: add another account.
-1: delete all accounts.
1
2019-01-18 09:05:40,413 - INFO - Logged-in successfully as '*****' using the cookie!

        1.Follow
        2.Like
        3.Comment
        4.Unfollow
        5.Block
        6.Setting
        7.Exit

What would you like to do?
2

        1. Like from hashtag(s)
        2. Like followers
        3. Like following
        4. Like last media likers
        5. Like our timeline
        6. Main menu

How do you want to like?
4

            1.Insert username
            2.Use username database

2
  0%|                                                    | 0/43 [00:00<?, ?it/s]2019-01-18 09:05:52,392 - INFO - Liking user_196425067's feed:
2019-01-18 09:05:52,821 - INFO - Going to like 2 medias.
                                                                               2019-01-18 09:05:52,972 - ERROR - Request returns 405 error! 0/2 [00:00<?, ?it/s]

Media ID error!
                                     

import getpass
import os
import random
import sys
import time


from tqdm import tqdm

# #sys.path.append(os.path.join(sys.path[0], '../'))
from instabot import Bot

bot = Bot(filter_business_accounts=False)
bot.login()

**Comment by maxdevblock** [comment]
@renevt 
1. did you edit
`/usr/lib/python2.7/site-packages/instabot/api/api.py`
```
def media_info(self, media_id):
    url = 'media/{media_id}/info/'.format(media_id=media_id)
    return self.send_request(url)
```

2. are you running the script with
`python multi_script_CLI.py`
or
`python3 multi_script_CLI.py`

**Comment by renevt** [comment]
@maxdevblock 

If you have Teamviewer and the time to help me
Can you help me out with this?

Im using a synology NAS 
i dont know to go to 
/usr/lib/python2.7/site-packages/instabot/api/api.py

im using python 2

**Comment by maxdevblock** [comment]
@renevt 
to edit the file simply
`sudo nano /usr/lib/python2.7/site-packages/instabot/api/api.py`
you'll have to type your password

**Comment by renevt** [comment]
@maxdevblock 

Thank you but the NAS is using a sort of linux so some command doesnt work . If do have really read a lot about linux. But what i try to say the nano command doesnt work. 
:/lib/python2.7/site-packages/instabot/api$ sudo nano /usr/lib/python2.7/site-packages/instabot/api/api.py
Password:
sudo: nano: command not found 

but so far i have locate the api file
:/lib/python2.7/site-packages/instabot/api$ ls
api_photo.py  api_photo.pyc  api.py  api.pyc  api_video.py  api_video.pyc  config.py  config.pyc  devices.py  devices.pyc  __init__.py  __init__.pyc  prepare.py  prepare.pyc

**Comment by maxdevblock** [comment]
@renevt 
remove your system installed instabot
```
pip uninstall instabot
```
and edit your local `instabot/api/api.py`
as we said.

don't forget to uncomment line
`# sys.path.append(os.path.join(sys.path[0], '../'))`
to original
`sys.path.append(os.path.join(sys.path[0], '../'))`
in your `multi_script_CLI.py`

**Comment by renevt** [comment]
@maxdevblock  

No i t doesnt work Max

Proceed (y/n)? administrator@DS216play:/$ pip uninstall instabot
Uninstalling instabot-0.4.5:
  Would remove:
    /usr/lib/python2.7/site-packages/instabot-0.4.5-py2.7.egg-info
    /usr/lib/python2.7/site-packages/instabot/*
Proceed (y/n)?
Your response ('administrator@ds216play:/$ pip uninstall instabot') was not one of the expected responses: y, n
Proceed (y/n)? Your response ('uninstalling instabot-0.4.5:') was not one of the expected responses: y, n
Proceed (y/n)? Your response ('would remove:') was not one of the expected responses: y, n
Proceed (y/n)? Your response ('/usr/lib/python2.7/site-packages/instabot-0.4.5-py2.7.egg-info') was not one of the expected responses: y, n
Proceed (y/n)? Your response ('/usr/lib/python2.7/site-packages/instabot/*') was not one of the expected responses: y, n
Proceed (y/n)? Your response ('proceed (y/n)?') was not one of the expected responses: y, n
Proceed (y/n)? y
Exception:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/pip/_internal/cli/base_command.py", line 143, in main
    status = self.run(options, args)
  File "/usr/lib/python2.7/site-packages/pip/_internal/commands/uninstall.py", line 75, in run
    auto_confirm=options.yes, verbose=self.verbosity > 0,
  File "/usr/lib/python2.7/site-packages/pip/_internal/req/req_install.py", line 683, in uninstall
    uninstalled_pathset.remove(auto_confirm, verbose)
  File "/usr/lib/python2.7/site-packages/pip/_internal/req/req_uninstall.py", line 224, in remove
    renames(path, new_path)
  File "/usr/lib/python2.7/site-packages/pip/_internal/utils/misc.py", line 280, in renames
    shutil.move(old, new)
  File "/usr/lib/python2.7/shutil.py", line 300, in move
    rmtree(src)
  File "/usr/lib/python2.7/shutil.py", line 252, in rmtree
    onerror(os.remove, fullname, sys.exc_info())
  File "/usr/lib/python2.7/shutil.py", line 250, in rmtree
    os.remove(fullname)
OSError: [Errno 13] Permission denied: '/usr/lib/python2.7/site-packages/instabot-0.4.5-py2.7.egg-info/dependency_links.txt'

 def media_info(self, media_id):
        url = 'media/{media_id}/info/'.format(media_id=media_id)
        return self.send_request(url)


import getpass
import os
import random
import sys
import time


from tqdm import tqdm

sys.path.append(os.path.join(sys.path[0], '../'))
from instabot import Bot

bot = Bot(filter_business_accounts=False)
bot.login()


Another ideas?

**Comment by maxdevblock** [comment]
@renevt 
create a new directory somewhere, cd to the dir and
```
git clone --single-branch --branch fix-issue-#756 https://github.com/maxdevblock/instabot.git
```
this will download my branch with the fix.
Try the multi_script_CLI that you find there in examples folder.

**Comment by kidd0ua** [comment]
@maxdevblock git clone --single-branch --branch fix-issue-#756 https://github.com/maxdevblock/instabot.git it works for me. Thanks!

**Comment by maxdevblock** [comment]
Yeppa :-)

**Comment by renevt** [comment]
No @maxdevblock it downt work either im really sorry i dont know whats goes wrong 
im feeling such a noob ...


administrator@DS216play://bot$ git clone --single-branch --branch fix-issue-#756 https://github.com/maxdevblock/instabot.git
-sh: git: command not found

administrator@DS216play://bot$ pip install  instabot.git
Collecting instabot.git
  Could not find a version that satisfies the requirement instabot.git (from versions: )
No matching distribution found for instabot.git
administrator@DS216play://bot$ ls
instabot.git

**Comment by maxdevblock** [comment]
@renevt 
```
apt-get install git
```

**Comment by ayush3298** [comment]
When we call check media, it sends request to 'media/{media_id}/info/'.format(media_id=media_id), but as it is returning error 405 that means the specified request HTTP method was received and recognized by the server, but the server has rejected that particular method for the requested resource. so we can guess Instagram has made some changes code is all good, but like all other endpoints, the API is working.
So the temporary solution is to skip checking for media until we found any other solution.

**Comment by maxdevblock** [comment]
> When we call check media, it sends request to 'media/{media_id}/info/'.format(media_id=media_id), but as it is returning error 405 that means the specified request HTTP method was received and recognized by the server, but the server has rejected that particular method for the requested resource. so we can guess Instagram has made some changes code is all good, but like all other endpoints, the API is working.
> So the temporary solution is to skip checking for media until we found any other solution.

You're right, but sending GET instead of POST actually works.

**Comment by iamalminko** [comment]
Thanks @maxdevblock, the fix worked for me. 
I'm running ubuntu 18.04, python 3.6.

**Comment by renevt** [comment]
@maxdevblock 

Ok did the  git server in the Nas

ok next problem Max

python multi_script_CLI.py
Traceback (most recent call last):
  File "multi_script_CLI.py", line 527, in <module>
    initial_checker()
  File "multi_script_CLI.py", line 23, in initial_checker
    with open(f, 'w') as f:
IOError: [Errno 13] Permission denied: 'hashtagsdb.txt'


Python3 
 python3  multi_script_CLI.py   Traceback (most recent call last):
  File "multi_script_CLI.py", line 7, in <module>
    from tqdm import tqdm
ImportError: No module named 'tqdm'

**Comment by maxdevblock** [comment]
@renevt 
1. for `ImportError: No module named 'tqdm'`
go to main cloned instabot directory (`cd ..` from _examples_ dir, if you're there) and
```
pip install -r requirements.txt
```

for `IOError: [Errno 13] Permission denied: 'hashtagsdb.txt'`
try
`sudo chown -R your_username:your_username *`
in the same directory. Obviously `your_username` is actually _your username_ :-)

**Comment by ohld** [comment]
```sudo chown -R your_username:your_username *```

This like looks like some bad thing to me. I don't think we need to ask users to do any `sudo`command. 

But if this will sole current problem for a specific case - ok.

**Comment by renevt** [comment]
@maxdevblock 

ok Max requirements.txt installed 

://bot/instabot/examples$  python  multi_script_CLI.py   Traceback (most recent call last):
  File "multi_script_CLI.py", line 527, in <module>
    initial_checker()
  File "multi_script_CLI.py", line 23, in initial_checker
    with open(f, 'w') as f:
IOError: [Errno 13] Permission denied: 'hashtagsdb.txt'

I do really appriciated your help ``

**Comment by maxdevblock** [comment]
@renevt 
try
```
chown your_username:your_username hashtagsdb.txt
```
in the dir where the file is

**Comment by silvio2810** [comment]
nothing, I've tried with all the fixing that all you posted here, and it doesn't work as well. @renevt @maxdevblock  I really appreciate your work

**Comment by maxdevblock** [comment]
@silvio2810 sorry about that. You have to wait for the fix to be merged and then
`pip install -U instabot`
to update your system installed instabot.
Keep an eye out for fix updating
#758

**Comment by renevt** [comment]
@maxdevblock 

i do really sorry max but i dont get out of it 
downloaded your fixed file but nothing change
still the same error

y cat
2019-01-18 12:35:38,283 - INFO - Instabot Started
Which account do you want to use? (Type number)
1: *****
0: add another account.
-1: delete all accounts.
1
2019-01-18 12:35:40,252 - INFO - Logged-in successfully as '******* the cookie!
2019-01-18 12:35:40,253 - INFO - Going to like media with hashtag #cat.
2019-01-18 12:35:42,612 - INFO - Going to like 85 medias.
  0%|                                                    | 0/85 [00:00<?, ?it/s]2019-01-18 12:35:42,757 - ERROR - Request returns 405 error!

Media ID error!

**Comment by maxdevblock** [comment]
@renevt same.
You have to wait for the fix to be merged and then
`pip install -U instabot`
to update your system installed instabot.
Keep an eye out for fix updating
#758

**Comment by renevt** [comment]
> pip install -U instabot

//volume1/web/instabot/examples$ pip install -U instabotRequirement already up-to-date: instabot in /usr/lib/python2.7/site-packages (0.4.5)
Requirement already satisfied, skipping upgrade: tqdm in /usr/lib/python2.7/site-packages (from instabot) (4.28.1)
Requirement already satisfied, skipping upgrade: requests-toolbelt in /usr/lib/python2.7/site-packages (from instabot) (0.8.0)
Requirement already satisfied, skipping upgrade: requests in /usr/lib/python2.7/site-packages (from instabot) (2.20.0)
Requirement already satisfied, skipping upgrade: schedule in /usr/lib/python2.7/site-packages (from instabot) (0.5.0)
Requirement already satisfied, skipping upgrade: future in /usr/lib/python2.7/site-packages (from instabot) (0.17.1)
Requirement already satisfied, skipping upgrade: six in /usr/lib/python2.7/site-packages (from instabot) (1.12.0)
Requirement already satisfied, skipping upgrade: huepy in /usr/lib/python2.7/site-packages (from instabot) (0.9.8.1)
Requirement already satisfied, skipping upgrade: certifi>=2017.4.17 in /usr/lib/python2.7/site-packages/certifi-2018.11.29-py2.7.egg (from requests->instabot) (2018.11.29)
Requirement already satisfied, skipping upgrade: chardet<3.1.0,>=3.0.2 in /usr/lib/python2.7/site-packages/chardet-3.0.4-py2.7.egg (from requests->instabot) (3.0.4)
Requirement already satisfied, skipping upgrade: idna<2.8,>=2.5 in /usr/lib/python2.7/site-packages (from requests->instabot) (2.7)
Requirement already satisfied, skipping upgrade: urllib3<1.25,>=1.21.1 in /usr/lib/python2.7/site-packages/urllib3-1.24.1-py2.7.egg (from requests->instabot) (1.24.1)
administrator@DS216play://volume1/web/instabot/examples$ python  like_hashtags.py cat
2019-01-18 12:37:54,640 - INFO - Instabot Started
Which account do you want to use? (Type number)
1: ****
0: add another account.
-1: delete all accounts.
1
2019-01-18 12:37:56,303 - INFO - Logged-in successfully as '****' using the cookie!
2019-01-18 12:37:56,306 - INFO - Going to like media with hashtag #cat.
2019-01-18 12:37:58,827 - INFO - Going to like 85 medias.
  0%|                                                    | 0/85 [00:00<?, ?it/s]2019-01-18 12:37:59,189 - ERROR - Request returns 405 error!

Media ID error!

Mabye something to do with the version of python?
I use the python and with python3 i get the error 
/instabot/examples$ python3  like_users.py Traceback (most recent call last):
  File "like_users.py", line 13, in <module>
    from instabot import Bot
  File "/volume1/web/instabot/examples/../instabot/__init__.py", line 1, in <module>
    from .api import API
  File "/volume1/web/instabot/examples/../instabot/api/__init__.py", line 1, in <module>
    from .api import API
  File "/volume1/web/instabot/examples/../instabot/api/api.py", line 15, in <module>
    import requests
ImportError: No module named 'request

**Comment by maxdevblock** [comment]
@renevt i say... **YOU HAVE TO WAIT** for the fix to get merged into master banch

**Comment by renevt** [comment]
> @renevt i say... **YOU HAVE TO WAIT** for the fix to get merged into master banch

Then i need to say thank you very much for your patient and help

Have good weekend

**Comment by ohld** [comment]
Not only merge but also pushed to pypi

**Comment by iamalminko** [comment]
For me autopost doesn't work well. I get an error with the current version, and strange thing is that the autopost in the fix-instabot works only once after installation and then it gets error. Anyone else with the same issue? This started happening at the same time when get_media error appeared.

**Comment by maxdevblock** [comment]
@alminkokasibovic autopost is totally another issue.

**Comment by iamalminko** [comment]
Is there a thread for autopost as well?

**Comment by maxdevblock** [comment]
Yes but I'm waiting for this issue to get fixed to update di PR

**Comment by Ritchie001** [comment]
Hello Max having the same problem got your branch Max ( open it in a new directory) but when i added my accounts 405 showed with liking (commenting does work!!!) 

So we'll have to wait then.

have a nice weekend

**Comment by imim332** [comment]
Installed instabot from the branch that had the fix.
Still doesn't work: [image](https://cdn.discordapp.com/attachments/535924033544388608/536148185156943892/unknown.png)

**Comment by maxdevblock** [comment]
@ImranDoet
did you previously install instabot via `pip`?
In this case, the imported package is the system one not the local, so the patch have no effect.

**Comment by f4depo** [comment]
I've download latest version of code but still giving 405 error... :/ [image](https://cdn.pbrd.co/images/HXaqVGA.png)

**Comment by maxdevblock** [comment]
To check if you got instabot installed in system directories, download, unzip and run this script
[check_instabot_756.py.zip](https://github.com/instagrambot/instabot/files/2775531/check_instabot_756.py.zip)

**Comment by f4depo** [comment]
> To check if you got instabot installed in system directories, download, unzip and run this script
> [check_instabot_756.py.zip](https://github.com/instagrambot/instabot/files/2775531/check_instabot_756.py.zip)

> C:\Users\Fers4t\Desktop>python check_instabot_756.py
>
> =====> INSTABOT FOUND IN YOUR SYSTEM DIRECTORIES
> /
>
> =====> CECHKING /api/api.py
> Traceback (most recent call last):
> File "check_instabot_756.py", line 14, in <module>
>    with open("{}api/api.py".format(instabot_dir), 'r') as f:
> FileNotFoundError: [Errno 2] No such file or directory: '/api/api.py'

**Comment by maxdevblock** [comment]
> > To check if you got instabot installed in system directories, download, unzip and run this script
> > [check_instabot_756.py.zip](https://github.com/instagrambot/instabot/files/2775531/check_instabot_756.py.zip)
> 
> > C:\Users\Fers4t\Desktop>python check_instabot_756.py
> 
> =====> INSTABOT FOUND IN YOUR SYSTEM DIRECTORIES
> /
> 
> =====> CECHKING /api/api.py
> Traceback (most recent call last):
> File "check_instabot_756.py", line 14, in
> with open("{}api/api.py".format(instabot_dir), 'r') as f:
> FileNotFoundError: [Errno 2] No such file or directory: '/api/api.py'

Will fix for windows, this script is for *Nix

**Comment by maxdevblock** [comment]
> > To check if you got instabot installed in system directories, download, unzip and run this script
> > [check_instabot_756.py.zip](https://github.com/instagrambot/instabot/files/2775531/check_instabot_756.py.zip)
> 
> > C:\Users\Fers4t\Desktop>python check_instabot_756.py
> > =====> INSTABOT FOUND IN YOUR SYSTEM DIRECTORIES
> > /
> > =====> CECHKING /api/api.py
> > Traceback (most recent call last):
> > File "check_instabot_756.py", line 14, in 
> > with open("{}api/api.py".format(instabot_dir), 'r') as f:
> > FileNotFoundError: [Errno 2] No such file or directory: '/api/api.py'

Anyhow, that means you've got instabot in your system directories. You have to wait the pip package to be update.

**Comment by f4depo** [comment]
> > > To check if you got instabot installed in system directories, download, unzip and run this script
> > > [check_instabot_756.py.zip](https://github.com/instagrambot/instabot/files/2775531/check_instabot_756.py.zip)
> > 
> > 
> > > C:\Users\Fers4t\Desktop>python check_instabot_756.py
> > 
> > 
> > =====> INSTABOT FOUND IN YOUR SYSTEM DIRECTORIES
> > /
> > =====> CECHKING /api/api.py
> > Traceback (most recent call last):
> > File "check_instabot_756.py", line 14, in
> > with open("{}api/api.py".format(instabot_dir), 'r') as f:
> > FileNotFoundError: [Errno 2] No such file or directory: '/api/api.py'
> 
> Will fix for windows, this script is for *Nix

Is there a working code or do i making bad someting? If I delete instabot from/with pip and run it manually, will does codes work?

**Comment by f4depo** [comment]
I've uninstalled instabot with pip and run codes manually but still doesn't work.. 

```
C:\Users\Fers4t\Desktop>python check_instabot_756.py

=====> ←[1;31mINSTABOT NOT FOUND IN YOUR SYSTEM DIRECTORIES←[0m
```

```
2019-01-19 12:55:58,841 - ERROR - Request returns 403 error!
2019-01-19 12:55:59,547 - ERROR - Request returns 404 error!
2019-01-19 12:55:59,547 - INFO - error, read exception bellow
2019-01-19 12:55:59,548 - INFO - Expecting value: line 1 column 1 (char 0)
```

**Comment by maxdevblock** [comment]
> I've uninstalled instabot with pip and run codes manually but still doesn't work..
> 
> ```
> C:\Users\Fers4t\Desktop>python check_instabot_756.py
> 
> =====> ←[1;31mINSTABOT NOT FOUND IN YOUR SYSTEM DIRECTORIES←[0m
> ```
> 
> ```
> 2019-01-19 12:55:58,841 - ERROR - Request returns 403 error!
> 2019-01-19 12:55:59,547 - ERROR - Request returns 404 error!
> 2019-01-19 12:55:59,547 - INFO - error, read exception bellow
> 2019-01-19 12:55:59,548 - INFO - Expecting value: line 1 column 1 (char 0)
> ```

Wait, working on it

**Comment by maxdevblock** [comment]
[check_instabot_756.py.zip](https://github.com/instagrambot/instabot/files/2775550/check_instabot_756.py.zip)
Try with this.

**Comment by f4depo** [comment]
I've uninstalled instaboth with pip... So no result. What should I do now?

```
C:\Users\Fers4t\Desktop>python check_instabot_756.py

=====> INSTABOT NOT FOUND IN YOUR SYSTEM DIRECTORIES


C:\Users\Fers4t\Desktop>
```

**Comment by maxdevblock** [comment]
> I've uninstalled instaboth with pip... So no result. What should I do now?
> 
> ```
> C:\Users\Fers4t\Desktop>python check_instabot_756.py
> 
> =====> INSTABOT NOT FOUND IN YOUR SYSTEM DIRECTORIES
> 
> 
> C:\Users\Fers4t\Desktop>
> ```

create a directory and
`git clone --single-branch --branch fix-issue-#756 https://github.com/maxdevblock/instabot.git`
Then try and example in examples folder you find there.

**Comment by imim332** [comment]
It's merged isnt it?

**Comment by maxdevblock** [comment]
> It's merged isnt it?

nope.

Look #758 still waiting for review.

**Comment by maxdevblock** [comment]
Everybody who installed via `pip` and applied the patch on their system installed instabot have solved the issue.
The fix in #758 is the solution.

**Comment by renevt** [comment]
@maxdevblock

Thank you for your hard work but i dont get it how to fix it. 
I rent a linux server since yesterday so i can use all the commands. 
Can you tell me stepp by stepp what i need to do please. ?
I understand install by pip but do not understand applied the patch.

**Comment by imim332** [comment]
Merged.

**Comment by imim332** [comment]
Still doesn't work for me btw, I used pip uninstall instabot, then pip install -U git+https://github.com/instagrambot/instabot.git
https://i.imgur.com/HhLwz66.png

**Comment by davidrsantos** [comment]
thanks

**Comment by renevt** [comment]
Ok guys it seems the bot is working ..

**Comment by ohld** [comment]
Very interesting discussion! Thanks to everyone that participated!

**Comment by imim332** [comment]
Still get a 405. After running pip uninstall instabot and then pip install instabot.

**Comment by ohld** [comment]
Try to install bleeding edge version from github:
``` bash
pip install -U git+https://github.com/instagrambot/instabot.git
```
